tensorboard/webapp/widgets/histogram/histogram_v2_test.ts

```diff
@@ -242,7 +237,131 @@ describe('histogram v2 test', () => {
 
         expect(
           getAxisLabelText(fixture.debugElement.query(byCss.Y_AXIS))
-        ).toEqual(['0.00', '200', '400', '600', '800', '1.00e+3', '1.20e+3']);
+        ).toEqual(['0.00', '500', '1.00e+3']);
+      });
+    });
+  });
+
+  describe('histogram render', () => {
+    function getGroupTransforms(element: DebugElement): string[] {
+      const transforms: string[] = [];
+      for (const debugEl of element.queryAll(By.css('.histogram'))) {
+        transforms.push(debugEl.attributes['transform']!);
+      }
+      return transforms;
+    }
+
+    function getHistogramPaths(element: DebugElement): string[] {
+      const pathD: string[] = [];
+      for (const debugEl of element.queryAll(By.css('path'))) {
+        pathD.push(debugEl.attributes['d']!);
+      }
+      return pathD;
+    }
+
+    describe('offset mode', () => {
+      it('positions group by their position in temporal axis', () => {
+        const fixture = createComponent('foo', [
+          buildHistogramDatum({step: 0}),
+          buildHistogramDatum({step: 5}),
+          buildHistogramDatum({step: 10}),
+        ]);
+        fixture.componentInstance.mode = HistogramMode.OFFSET;
+        fixture.componentInstance.timeProperty = TimeProperty.STEP;
+        fixture.detectChanges();
+
+        expect(
+          getGroupTransforms(fixture.debugElement.query(byCss.HISTOGRAMS))
+        ).toEqual([
+          // The content box is 30x50 pixels and because we need to render 2.5D
+          // on 2D screen, we give height / 2.5 space for histogram slices to
+          // render. 50 / 2.5 = 20 so effectively, step=0 is rendered at 20px
+          // from the top.
+          'translate(0, 20)',
+          'translate(0, 35)',
+          'translate(0, 50)',
+        ]);
+      });
+
+      it('moves the group position depending on the timeProperty', () => {
+        const fixture = createComponent('foo', [
+          buildHistogramDatum({step: 0, wallTime: 100}),
+          buildHistogramDatum({step: 5, wallTime: -200}),
+          buildHistogramDatum({step: 10, wallTime: 400}),
+        ]);
+        fixture.componentInstance.mode = HistogramMode.OFFSET;
+        fixture.componentInstance.timeProperty = TimeProperty.STEP;
```

Comment:
Is it intentional to set the timeProperty to STEP when L296 changes it again, without a test expectation in between?

Reply:
Yup,k I wanted to assert whether it updates correctly. We already have test spec for asserting on initial render with `STEP` so I omitted that.